libsystemd-sys/build.rs

```diff
@@ -2,21 +2,41 @@ extern crate pkg_config;
 use std::env;
 
 fn main() {
-    let e = match pkg_config::find_library("libsystemd") {
+    #[cfg(not(feature = "elogind"))]
+    let library = pkg_config::find_library("libsystemd");
+    #[cfg(feature = "elogind")]
+    let library = pkg_config::find_library("libelogind");
+
+    let error = match library {
         Ok(_) => return,
-        Err(e) => e,
+        Err(error) => error,
     };
 
-    match env::var("LIBSYSTEMD_LDFLAGS") {
+    #[cfg(not(feature = "elogind"))]
+    let ld_preload_var = "LIBSYSTEMD_LDFLAGS";
+    #[cfg(feature = "elogind")]
+    let ld_preload_var = "LIBELOGIND_LDFLAGS";
+
+    match env::var(ld_preload_var) {
         Ok(flags) => {
             /* Ideally we'd avoid rustc-flags in favor of rustc-link-{search,lib}, but this should
              * work fine
              */
-            println!("cargo:rustc-flags={}", flags);
+            eprintln!("cargo:rustc-flags={}", flags);
```

Comment:
I'm pretty sure these need to be `println()` so that they're picked up by cargo?